src/index.js

```diff
@@ -257,51 +248,45 @@ function withConnection (server, work, cancellable) {
   return promise
 }
 
-function connect (server) {
+function getApplicationName () {
+  let path = findRoot(process.argv[1] || process.cwd()) + '/package.json'
+  let pkg = JSON.parse(readFileSync(path, 'utf8'))
+  return pkg.name
+}
+
+function configure (server) {
   if (typeof server === 'string') {
     server = parseConnectionString(server)
   } else if (typeof server === 'undefined') {
     server = {}
   }
 
-  server.poolSize = server.poolSize || process.env.PG_POOL_SIZE
-  server.poolIdleTimeout = (
-    server.poolIdleTimeout ||
+  server.max = server.poolSize || process.env.PG_POOL_SIZE
+  server.idleTimeoutMillis = (
+    server.idleTimeoutMillis ||
     process.env.PG_IDLE_TIMEOUT ||
     (process.env.NODE_ENV === 'test' && 1)
   )
-  server.reapIntervalMillis = (
-    server.reapIntervalMillis ||
-    process.env.PG_REAP_INTERVAL ||
-    (process.env.NODE_ENV === 'test' && 50)
-  )
   server.application_name = (
     server.application_name ||
     process.env.APPLICATION_NAME ||
     getApplicationName()
   )
 
-  return new Promise(function doConnection (resolve, reject) {
-    return pg.connect(server, function onConnect (err, client, done) {
-      if (err) {
-        reject(err)
-      } else {
-        resolve([client, done])
-      }
-    })
-  })
-}
+  let pool = new Pool(server)
 
-function getApplicationName () {
-  var path = findRoot(process.argv[1] || process.cwd()) + '/package.json'
-  var pkg = JSON.parse(readFileSync(path, 'utf8'))
-  return pkg.name
-}
+  function connect () {
+    return new Promise((resolve, reject) => {
```

Comment:
I'm reasonably sure node-postgres moved to using promises everywhere, but this is here for extra paranoia